fecgov/fec-cms
# Case subjects filter MURs

## Summary (required)

- Resolves #6500
   - also #6503
   - also #6538

Issue 6500:
   - One single select for Case Subject (primary_subject_id)
   - One single select for Secondary Case Subject (secondary_subject_id) that is only shown and populated when the chosen case subject has secondary subjects
   - Set pagination back to page one upon filter-tag removal in case there are fewer pages in resulting data (otherwise it shows a blank table, [example on prod](https://www.fec.gov/data/legal/search/murs/?search=&case_respondents=&case_min_open_date=&case_max_open_date=&case_min_close_date=&case_max_close_date=&offset=140&limit=20&sort=&mur_disposition_category_id=6#results-murs))
   - Alterations to the dropdown_json macro to make it more versatile
   - Alterations to the constants for Audit search to make it work with the altered macro
 
Issue 6503:
 - Add tooltip to macro and subject filters in MUR template
 -  Change language  in `Case details and documents` column for  archived MURs
 
 Issue 6538:
 - remove un needed disposition categories from constants

### Required reviewers

Required: one frontend
Optional: One UX
@fec-jli - Just to check Audit search

## Impacted areas of the application

MUR and ADR detatables

## Screenshots

(Include a screenshot of the new/updated features in context (“in the wild”). If it is an interface change, include both before and after screenshots)


## How to test

- `npm run build-sass`
- Test Case subjects filter
- Filter tag removal: 
    - You can remove a secondary subject filter tag to filter only the primary subject for different results
         - Example: remove `Corporations` from `Contributions- Corporations`
    - When you remove a primary filter tag that has a secondary subject filter tag too, they are both removed together
         - Example: there is no difference in results between filtering  `Contributions- Corporations` and  `Corporations` because that secondary subject is tied to the primary subject.
         - There is one edge case where the  secondary subject `Exemptions`,  exists under two primary subjects (`Contributions` and `Expenditures`). So when adding  `Exemptions` as a secondary subject to either , results include both. We can address this later if necessary.
- Test pagination 
- Confirm new language  in `Case details and documents` column for archived MURs (issue 6503)
- @fec-jli - Please do a cursory check of  Audit search `Category` and `Sub-category` selects as there was an adjustment  to the constants that build the Category select for this PR.
